### PR TITLE
feat: add user invite management

### PR DESCRIPTION
**Pull request by Meierschlumpf**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

### CODE REVIEW

File: packages/api/src/router/invite.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        creator: {
+          columns: {
+            id: true,
+            name: true,
+          },
+        },
+      },
+    });
+    return dbInvites;
+  }),
```
Comment (SeDemal) [inline packages/api/src/router/invite.ts:27]: Aren't we missing the "byId" procedure?
Or is the check done by going against all existing invites?

Comment (Meierschlumpf) [reply packages/api/src/router/invite.ts:27]: It's not yet implemented that you can join Homarr with an invite url. This is only the management of it

File: apps/nextjs/src/app/[locale]/manage/users/invites/_components/invite-copy-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+export const InviteCopyModal = createModal<RouterOutputs["invite"]["create"]>(
+  ({ actions, innerProps }) => {
+    const t = useScopedI18n("management.page.user.invite");
+    const inviteUrl = useInviteUrl(innerProps);
+
+    return (
+      <Stack>
+        <Text>{t("action.copy.description")}</Text>
+        {/* FIXME: When next-international v2 is released the descriptions bold element can be implemented, see https://github.com/QuiiBz/next-international/pull/361 for progress */}
```
Comment (manuel-rw) [inline apps/nextjs/src/app/[locale]/manage/users/invites/_components/invite-copy-modal.tsx:17]: Rename to TODO?

File: packages/api/src/router/invite.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import { TRPCError } from "@trpc/server";
+
+import { asc, createId, eq } from "@homarr/db";
+import { invites } from "@homarr/db/schema/sqlite";
+import { z } from "@homarr/validation";
+
+import { createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const inviteRouter = createTRPCRouter({
+  all: protectedProcedure.query(async ({ ctx }) => {
```
Comment (manuel-rw) [inline packages/api/src/router/invite.ts:11]: Action is unclear. Rename to ``getAll``

File: packages/api/src/router/invite.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          columns: {
+            id: true,
+            name: true,
+          },
+        },
+      },
+    });
+    return dbInvites;
+  }),
+  create: protectedProcedure
```
Comment (manuel-rw) [inline packages/api/src/router/invite.ts:28]: Rename to ``createInvite``

File: packages/api/src/router/invite.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        creatorId: ctx.session.user.id,
+        token,
+      });
+
+      return {
+        id,
+        token,
+      };
+    }),
+  delete: protectedProcedure
```
Comment (manuel-rw) [inline packages/api/src/router/invite.ts:50]: Rename to ``deleteBoard``

**Review by manuel-rw — CHANGES_REQUESTED** [review]

**Comment by Meierschlumpf** [comment]
Regarding the renaming of those router endpoints. I understand why you want to change it but on the same time it is kind of unnecessary as you'll always call them with `clientApi.invite.create` which already contains what you create and so it's kind of a duplicate

**Comment by manuel-rw** [comment]
> Regarding the renaming of those router endpoints. I understand why you want to change it but on the same time it is kind of unnecessary as you'll always call them with `clientApi.invite.create` which already contains what you create and so it's kind of a duplicate

No, this is not my point. Consider the following endpoints: ``clientApi.invite.create`` and ``clientApi.invite.createTemporaryToken`` (note: these are illustrative names / ideas). So what does the first one do? Yes, the ``invite`` router does indicate it roughly, but since there are now to names, it is no longer clear.

**Review by manuel-rw — APPROVED** [review]

**Pull request merged**